test/apps/iodemo/io_demo.cc

```diff
@@ -157,12 +203,18 @@ class P2pDemoCommon : public UcxContext {
 
     bool send_io_message(UcxConnection *conn, io_op_t op,
                          uint32_t sn, size_t data_size) {
-        IoMessage *m = new IoMessage(opts().iomsg_size, op, sn,
-                                     data_size);
+        IoMessage *m = ioMsgPool.Allocate();
+        m->init(opts().iomsg_size, op, sn, data_size);
         VERBOSE_LOG << "sending IO " << io_op_names[op] << ", sn " << sn
                     << " data size " << data_size;
         return conn->send_io_message(m->buffer(), opts().iomsg_size, m);
     }
+    
+public:
+    void static freeIoMsgPool(IoMessage* iomsg) {
+        ioMsgPool.Free(iomsg);
+    }
+    static MemoryPool<IoMessage> ioMsgPool;
```

Comment:
1. why static and not regular member variable?
2. rename to _io_msg_pool

Reply:
1. If don't use static then we should pass the pool object address to the item class to free it. I add another parameter for this in init() function for this and change to regular member variables. Let me know please if any other suggestion. Thanks!

Reply:
2. Fixed.